src/keymap.rs

```diff
@@ -0,0 +1,63 @@
+use glib::translate::*;
+use std::mem;
+use std::ptr;
+use Keymap;
+use KeymapKey;
+
+impl Keymap {
+    pub fn get_entries_for_keycode(
+        &self,
+        hardware_keycode: u32,
+    ) -> Option<(Vec<KeymapKey>, Vec<u32>)> {
+        unsafe {
+            let mut keys = ptr::null_mut();
+            let mut keyvals = ptr::null_mut();
+            let mut n_entries = mem::MaybeUninit::uninit();
+            let ret = from_glib(gdk_sys::gdk_keymap_get_entries_for_keycode(
+                self.to_glib_none().0,
+                hardware_keycode,
+                &mut keys,
+                &mut keyvals,
+                n_entries.as_mut_ptr(),
+            ));
+            if ret {
+                let n_entries = n_entries.assume_init() as usize;
+                let mut r_keys = Vec::with_capacity(n_entries);
+                let mut r_keyvals = Vec::with_capacity(n_entries);
+                for i in 0..n_entries {
+                    r_keys.push(from_glib_none(keys.add(i)));
+                    r_keyvals.push(ptr::read(keyvals.add(i)));
```

Comment:
This seems strange. Why `(Vec<KeymapKey>, Vec<u32>)` and not `Vec<(KeymapKey, u32)>` if they're the same size anyway and apparently belong together? :)

Reply:
Just thought about it afterwards. :)